### ISSUE TITLE
linter: `react/jsx-key` false positive when `key` is inside a spread object

### ISSUE THREAD

**Issue by jeanne-cai** [oxc-project/oxc#23973]
### What version of Oxlint are you using?

1.70.0

### What command did you run?

`yarn oxlint -c oxlint.config.mjs`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```ts
import { defineConfig } from 'oxlint';

export default defineConfig({
    categories: {
        correctness: 'off',
    },
    plugins: ['react'],
    rules: {
        'react/jsx-key': 'error',
    },
});
```

### What happened?

`key` is present in the spread object on the line above the return. ESLint does not flag this pattern.

test.tsx
```tsx
const items = [
    { id: 'a', type: 'foo' },
    { id: 'b', type: 'bar' },
];

const Test = () => (
    <>
        {items.map((item) => {
            const props = { key: item.id, className: 'x' };
            switch (item.type) {
                case 'foo':
                    return <div {...props}>foo</div>;
                default:
                    return <span {...props}>bar</span>;
            }
        })}
    </>
);
```

Output
```
  × react(jsx-key): Missing "key" prop for element in iterator.
    ╭─[test.tsx:8:16]
  7 │     <>
  8 │         {items.map((item) => {
    ·                ─┬─
    ·                 ╰── Iterator starts here.
  9 │             const props = { key: item.id, className: 'x' };
 10 │             switch (item.type) {
 11 │                 case 'foo':
 12 │                     return <div {...props}>foo</div>;
    ·                             ─┬─
    ·                              ╰── Element generated here.
 13 │                 default:
    ╰────
  help: Add a "key" prop to the element in the iterator (https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key).

  × react(jsx-key): Missing "key" prop for element in iterator.
    ╭─[test.tsx:8:16]
  7 │     <>
  8 │         {items.map((item) => {
    ·                ─┬─
    ·                 ╰── Iterator starts here.
  9 │             const props = { key: item.id, className: 'x' };
 10 │             switch (item.type) {
 11 │                 case 'foo':
 12 │                     return <div {...props}>foo</div>;
 13 │                 default:
 14 │                     return <span {...props}>bar</span>;
    ·                             ──┬─
    ·                               ╰── Element generated here.
 15 │             }
    ╰────
  help: Add a "key" prop to the element in the iterator (https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key).
```

**Comment by camc314** [comment]
Hi! Thanks for reporting this issue

I took a look at the upstream behaviour here, and it looks like this is a case where we check more deeply than the upstream rule.

The upstream rule does not appear to collect `return` stmts nested inside `switch` cases, so th example is not reported there because the JSX return is effectively skipped. If the same spread pattern is moved out of the `switch`, for example:

```tsx
items.map((item) => {
  const props = { key: item.id, className: 'x' };
  return <div {...props}>foo</div>;
});
```

then upstream reports `Missing "key" prop for element in iterator` as well.

So the difference is not that upstream understands `key` inside the spread object. The difference is that Oxc detects JSX returned from `switch` cases, which is stricter and arguably better behavior. We should keep that deeper detection and fix this as a spread/object-literal false positive rather than matching upstream’s `switch` blind spot.

Closing this as not planned as our behaviour is better than the upstream rule.

Let me know what you think - thanks!

**Comment by jeanne-cai** [comment]
Hi! I understand the reasoning. Thanks for the explanation!
